services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/containers.py

```diff
@@ -0,0 +1,241 @@
+import logging
+import traceback
+
+# pylint: disable=redefined-builtin
+from typing import Any, Dict, List, Union
+
+import aiodocker
+from fastapi import (
+    APIRouter,
+    BackgroundTasks,
+    Depends,
+    HTTPException,
+    Query,
+    Request,
+    status,
+)
+from fastapi.responses import PlainTextResponse
+
+from ..core.dependencies import get_settings, get_shared_store
+from ..core.settings import DynamicSidecarSettings
+from ..core.shared_handlers import remove_the_compose_spec, write_file_and_run_command
+from ..core.utils import assemble_container_names, docker_client
+from ..core.validation import InvalidComposeSpec, validate_compose_spec
+from ..models.domains.shared_store import SharedStore
+
+logger = logging.getLogger(__name__)
+
+containers_router = APIRouter(tags=["containers"])
+
+
+async def task_docker_compose_up(
+    settings: DynamicSidecarSettings,
+    shared_store: SharedStore,
+) -> None:
+    # building is a security risk hence is disabled via "--no-build" parameter
+    command = (
+        "docker-compose --project-name {project} --file {file_path} "
+        "up --no-build --detach"
+    )
+    finished_without_errors, stdout = await write_file_and_run_command(
+        settings=settings,
+        file_content=shared_store.compose_spec,
+        command=command,
+        command_timeout=None,
+    )
+    message = f"Finished {command} with output\n{stdout}"
+
+    if finished_without_errors:
+        logger.info(message)
+    else:
+        logger.error(message)
+
+    return None
+
+
+def _raise_if_container_is_missing(id: str, container_names: List[str]) -> None:
+    if id not in container_names:
+        message = (
+            f"No container '{id}' was started. Started containers '{container_names}'"
+        )
+        logger.warning(message)
+        raise HTTPException(status.HTTP_404_NOT_FOUND, detail=message)
+
+
+def _raise_from_docker_error(error: aiodocker.exceptions.DockerError) -> None:
+    logger.warning(
+        "An unexpected Docker error occurred:\n%s", str(traceback.format_exc())
+    )
+    raise HTTPException(
+        status.HTTP_500_INTERNAL_SERVER_ERROR, detail=error.message
```

Comment:
isn't it a better code for this? As I mentioned in previous reviews, the default of fastapi is actually 500. So e.g. if you would not have caught the Docker error exception, it would have already responded the same way.

Reply:
Refactored: moved inside the docker_client() context manager. Where the exception will be logged (to help with debugging. The client will still receive a 500 with a description that there is something wrong with Docker.